forge-game/src/main/java/forge/game/ability/effects/ChangeZoneEffect.java

```diff
@@ -516,10 +516,15 @@ private void changeKnownOriginResolve(final SpellAbility sa) {
                 final CardCollection random = new CardCollection(tgtCards);
                 CardLists.shuffle(random);
                 tgtCards = random;
-            } else if (sa.hasParam("Chooser")) {
-                tgtCards = chooser.getController().orderMoveToZoneList(tgtCards, destination, sa);
             } else {
-                tgtCards = GameActionUtil.orderCardsByTheirOwners(game, tgtCards, destination, sa);
+                if (sa.hasParam("Chooser"))
+                    tgtCards = chooser.getController().orderMoveToZoneList(tgtCards, destination, sa);
+                else
+                    tgtCards = GameActionUtil.orderCardsByTheirOwners(game, tgtCards, destination, sa);
+
+                final CardCollection reversed = new CardCollection(tgtCards);
+                Collections.reverse(reversed);
+                tgtCards = reversed;
```

Comment:
@tool4ever i think the extra Reverse seems a bit ugly ... can we do something about this?

Reply:
Wasn't too happy with that part either, but I couldn't find an elegant way to reverse a CardCollectionView.

Reply:
@tool4ever probably needs to look into the `orderMoveToZoneList` and `orderCardsByTheirOwners` functions, that for destination=Library already reverses the CardCollection

Reply:
Since this seems more of a convenience thing to select the order opposite from the zone changes it should probably belong to GUI code (PlayerControllerHuman)
or just switch the "bottom" + "top" labels around? :p
either way then all places in effect classes could be cleaned up

Reply:
It's a question of whether you want the output of `orderMoveToZoneList` to mean "the player wants the cards in this order" versus "this is the order you should move cards one-by-one into this zone and position". Whatever is settled on, it would probably be good idea to add a Javadoc comment explaining the function's behavior when ordering cards for the top of a deck.

Reply:
Second one makes more sense?
So all loops can be for each

Reply:
i don't know, we probably need more info like @tehdiplomat

Reply:
Ideally, we would have a different UI element thats a Scry type. Where you can reorder a library AND send to a different zone (including bottom of the library). I know people have had similar issues with Sylvan Library, where the ordering and the place on library didn't up matching their expectations. It seems weird to reverse the array when we already have labels to say which order things will be in.